packages/ckeditor5-image/src/image/converters.ts

```diff
@@ -249,9 +249,18 @@ export function downcastSourcesAttribute( imageUtils: ImageUtils ): ( dispatcher
 				viewElement = parentElement;
 			}
 
-			// Insert the picture and move img into it.
-			viewWriter.insert( viewWriter.createPositionBefore( imgElement ), pictureElement );
-			viewWriter.move( viewWriter.createRangeOn( imgElement ), viewWriter.createPositionAt( pictureElement, 'end' ) );
+			const prevPictureElement = imgElement.parent!.is( 'element', 'picture' ) ? imgElement.parent : null;
+
+			if ( prevPictureElement ) {
+				// Insert a new picture, move img into it, remove previous picture element (ckeditor5#17192).
+				viewWriter.insert( viewWriter.createPositionBefore( prevPictureElement ), pictureElement );
+				viewWriter.move( viewWriter.createRangeOn( imgElement ), viewWriter.createPositionAt( pictureElement, 'end' ) );
+				viewWriter.remove( prevPictureElement );
```

Comment:
What would happen if other converters downcast other attributes to the picture element? Would those be lost? Maybe in such a case, we should remove source elements from the picture element and reinsert them into the already existing picture element.

Reply:
Maybe not emptying but removing source elements and inserting new ones (img element should also be reused)

Reply:
Fixed in https://github.com/ckeditor/ckeditor5/pull/17193/commits/893b1e307a0b2d9446f20d1f1aebd0d26f49a361 and even the code go slightly simpler 👍 I had to add a custom attribute handling in test because we don't have anything for picture from what I see.